### PR TITLE
feat: support removing kernel args

### PR DESCRIPTION
**Pull request by frezbo**
Support removing kernel args that start with a `-`.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/siderolabs/go-procfs
 
-go 1.21
+go 1.21.0
+
+toolchain go1.21.4
```
Comment (smira) [inline go.mod]: you can better do `go 1.21.4`

File: procfs/cmdline.go [hunk 2]
```diff
@@ -184,6 +184,40 @@ func (c *Cmdline) SetAll(args []string) {
...
+func (c *Cmdline) Delete(v *Parameter) {
+	c.Lock()
+	defer c.Unlock()
+
+	for i, value := range c.Parameters {
+		if value.key == v.key {
+			for _, val := range v.values {
+				for j, v := range value.values {
+					if v == val {
+						c.Parameters[i].values = append(c.Parameters[i].values[:j], c.Parameters[i].values[j+1:]...)
```
Comment (smira) [inline procfs/cmdline.go]: we could make it less weird looking by using e.g. https://pkg.go.dev/slices#Delete

Comment (smira) [reply procfs/cmdline.go]: or even `DeleteFunc` there

File: procfs/cmdline.go [hunk 3]
```diff
@@ -184,6 +184,40 @@ func (c *Cmdline) SetAll(args []string) {
...
 }
 
+// Delete deletes a kernel parameter by exact match.
+func (c *Cmdline) Delete(v *Parameter) {
+	c.Lock()
+	defer c.Unlock()
+
+	for i, value := range c.Parameters {
+		if value.key == v.key {
+			for _, val := range v.values {
```
Comment (smira) [inline procfs/cmdline.go]: if we drop all values, should we drop the outer key?

I mean like we had

`console=abc`

and I append with `-console=abc`, should it drop `console` key completely now?

**Review by smira — APPROVED** [review]

File: procfs/cmdline.go (hunk 3 above)
Comment (frezbo) [reply procfs/cmdline.go]: i didn't quite get that, this maybe: https://github.com/siderolabs/go-procfs/pull/11/files#diff-1f6502d8a4e4009fba4aa162aba1cb59dca996f3dee72949ac177e56fc0185b0R136-R138

Comment (smira) [reply procfs/cmdline.go]: yep, looks like it works in the tests, I meant dropping (now empty) value from c.Parameters

Comment (smira) [reply procfs/cmdline.go]: not sure if it's important even

File: procfs/cmdline.go (hunk 2 above)
Comment (frezbo) [reply procfs/cmdline.go]: oh, this is cool

File: procfs/cmdline.go (hunk 3 above)
Comment (frezbo) [reply procfs/cmdline.go]: I guess it's fine

**Pull request merged**